pkg/common/cns-lib/volume/manager.go

```diff
@@ -136,10 +136,16 @@ var (
 	// read/write on manager instance.
 	managerInstanceLock sync.Mutex
 	volumeTaskMap       = make(map[string]*createVolumeTaskDetails)
+	snapshotTaskMap     = make(map[string]*createSnapshotTaskDetails)
 	// Alias for CreateVolumeOperationRequestDetails function declaration.
 	createRequestDetails = cnsvolumeoperationrequest.CreateVolumeOperationRequestDetails
 )
 
+// createSnapshotTaskDetails has the same structure as createVolumeTaskDetails
+type createSnapshotTaskDetails struct {
+	createVolumeTaskDetails
```

Comment:
Does reusing this cause any problem for the create volume code path?

Reply:
I don't think so. It is just type reusing not object reusing. I could have used the `createVolumeTaskDetails` directly, but I think it would be confusing to use `createVolumeTaskDetails` in `CreateSnapshot` implementation. Meanwhile, the regression test works just fine.